// @catho/quantum/Grid/sub-components/Hide.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _media = require("./shared/media");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Hide = _styledComponents.default.div.withConfig({
  displayName: "Hide",
  componentId: "sc-13m0had-0"
})(["", " ", " ", " ", " ", ""], function (props) {
  return props.xlarge && _media.hideQueries.xlarge();
}, function (props) {
  return props.large && _media.hideQueries.large();
}, function (props) {
  return props.medium && _media.hideQueries.medium();
}, function (props) {
  return props.small && _media.hideQueries.small();
}, function (props) {
  return props.xsmall && _media.hideQueries.xsmall();
});

Hide.propTypes = {
  xlarge: _propTypes.default.bool,
  large: _propTypes.default.bool,
  medium: _propTypes.default.bool,
  small: _propTypes.default.bool,
  xsmall: _propTypes.default.bool
};
Hide.displayName = 'Hide';
var _default = Hide;
exports.default = _default;
