// @catho/quantum-storybook-ui/dist/components/Example/Example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../../ui/Colors"));

var _CodeExample = _interopRequireDefault(require("../CodeExample"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  margin-bottom: 20px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  cursor: pointer;\n  text-transform: lowercase;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Small = _styledComponents.default.small(_templateObject(), _Colors.default.grey.harder);

var Code = _styledComponents.default.div(_templateObject2());

var Example = function Example(_ref) {
  var Component = _ref.component,
      title = _ref.title,
      importModules = _ref.importModules,
      code = _ref.code;
  return _react.default.createElement(_react.default.Fragment, null, Component, _react.default.createElement("h3", null, title), _react.default.createElement(Code, null, _react.default.createElement(_CodeExample.default, {
    code: code,
    withImport: importModules,
    showTitle: false,
    component: Component
  })));
};

Example.defaultProps = {
  code: '',
  title: '',
  importModules: '',
  component: {}
};
Example.propTypes = {
  code: _propTypes.default.string,
  component: _propTypes.default.instanceOf(Object),
  title: _propTypes.default.string,
  importModules: _propTypes.default.string
};
var _default = Example;
exports.default = _default;
