// @catho/quantum/shared/shadow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _hexToRgba = _interopRequireDefault(require("./hexToRgba"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SHADOWS = {
  UMBRA: {
    elevations: ['0px 0px 0px 0px', '0px 2px 1px -1px', '0px 3px 1px -2px', '0px 3px 3px -2px', '0px 2px 4px -1px', '0px 3px 5px -1px', '0px 3px 5px -1px', '0px 4px 5px -2px', '0px 5px 5px -3px', '0px 5px 6px -3px', '0px 6px 6px -3px', '0px 6px 7px -4px', '0px 7px 8px -4px', '0px 7px 8px -4px', '0px 7px 9px -4px', '0px 8px 9px -5px', '0px 8px 10px -5px', '0px 8px 11px -5px', '0px 9px 11px -5px', '0px 9px 12px -6px', '0px 10px 13px -6px', '0px 10px 13px -6px', '0px 10px 14px -6px', '0px 11px 14px -7px', '0px 11px 15px -7px'],
    opacity: 0.2
  },
  PENUMBRA: {
    elevations: ['0px 0px 0px 0px', '0px 1px 1px 0px', '0px 2px 2px 0px', '0px 3px 4px 0px', '0px 4px 5px 0px', '0px 5px 8px 0px', '0px 6px 10px 0px', '0px 7px 10px 1px', '0px 8px 10px 1px', '0px 9px 12px 1px', '0px 10px 14px 1px', '0px 11px 15px 1px', '0px 12px 17px 2px', '0px 13px 19px 2px', '0px 14px 21px 2px', '0px 15px 22px 2px', '0px 16px 24px 2px', '0px 17px 26px 2px', '0px 18px 28px 2px', '0px 19px 29px 2px', '0px 20px 31px 3px', '0px 21px 33px 3px', '0px 22px 35px 3px', '0px 23px 36px 3px', '0px 24px 38px 3px'],
    opacity: 0.14
  },
  AMBIENT: {
    elevations: ['0px 0px 0px 0px', '0px 1px 3px 0px', '0px 1px 5px 0px', '0px 1px 8px 0px', '0px 1px 10px 0px', '0px 1px 14px 0px', '0px 1px 18px 0px', '0px 2px 16px 1px', '0px 3px 14px 2px', '0px 3px 16px 2px', '0px 4px 18px 3px', '0px 4px 20px 3px', '0px 5px 22px 4px', '0px 5px 24px 4px', '0px 5px 26px 4px', '0px 6px 28px 5px', '0px 6px 30px 5px', '0px 6px 32px 5px', '0px 7px 34px 6px', '0px 7px 36px 6px', '0px 8px 38px 7px', '0px 8px 40px 7px', '0px 8px 42px 7px', '0px 9px 44px 8px', '0px 9px 46px 8px'],
    opacity: 0.12
  }
};
var _SHADOWS$UMBRA = SHADOWS.UMBRA,
    umbraElevations = _SHADOWS$UMBRA.elevations,
    umbraOpacity = _SHADOWS$UMBRA.opacity,
    _SHADOWS$PENUMBRA = SHADOWS.PENUMBRA,
    penumbraElevations = _SHADOWS$PENUMBRA.elevations,
    penumbraOpacity = _SHADOWS$PENUMBRA.opacity,
    _SHADOWS$AMBIENT = SHADOWS.AMBIENT,
    ambientElevations = _SHADOWS$AMBIENT.elevations,
    ambientOpacity = _SHADOWS$AMBIENT.opacity;

var shadowWithTheme = function shadowWithTheme(_ref, elevation) {
  var defaultColor = _ref.colors.neutral[700];
  var color = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultColor;
  var umbra = "".concat(umbraElevations[elevation], " ").concat((0, _hexToRgba.default)(color, umbraOpacity));
  var penumbra = "".concat(penumbraElevations[elevation], " ").concat((0, _hexToRgba.default)(color, penumbraOpacity));
  var ambient = "".concat(ambientElevations[elevation], " ").concat((0, _hexToRgba.default)(color, ambientOpacity));
  return "box-shadow: ".concat(umbra, ", ").concat(penumbra, ", ").concat(ambient, ";");
};

var shadow = function shadow() {
  var elevation = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 0;
  var color = arguments.length > 1 ? arguments[1] : undefined;
  return function (_ref2) {
    var theme = _ref2.theme;
    return shadowWithTheme(theme, elevation, color);
  };
};

var _default = shadow;
exports.default = _default;
