// @catho/quantum/Tooltip/options.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var ARROW_SIZE = 6;
var PERCENTAGE_Y = 130;
var PERCENTAGE_X = 50;
var upDownBorders = "\n  border-left: ".concat(ARROW_SIZE, "px solid transparent;\n  border-right: ").concat(ARROW_SIZE, "px solid transparent;\n  left: 50%;\n  transform: translateX(-50%);\n");
var sideBorders = "\n  border-top: ".concat(ARROW_SIZE, "px solid transparent;\n  border-bottom: ").concat(ARROW_SIZE, "px solid transparent;\n  top: 50%;\n  transform: translateY(-50%);\n");
var placementConfig = {
  arrowPosition: {
    top: "\n      ".concat(upDownBorders, "\n      border-top: ").concat(ARROW_SIZE, "px solid;\n      border-top-color: inherit;\n      bottom: -").concat(ARROW_SIZE - 1, "px;\n    "),
    right: "\n    ".concat(sideBorders, "\n    border-right: ").concat(ARROW_SIZE, "px solid;\n    border-right-color: inherit;\n    left: -").concat(ARROW_SIZE - 1, "px;\n  "),
    left: "\n      ".concat(sideBorders, "\n      border-left: ").concat(ARROW_SIZE, "px solid;\n      border-left-color: inherit;\n      right: -").concat(ARROW_SIZE - 1, "px;\n    "),
    bottom: "\n      ".concat(upDownBorders, "\n      border-bottom: ").concat(ARROW_SIZE, "px solid;\n      border-bottom-color: inherit;\n      top: -").concat(ARROW_SIZE - 1, "px;\n    ")
  },
  tipPosition: {
    top: "\n      left: ".concat(PERCENTAGE_X, "%;\n      bottom: ").concat(PERCENTAGE_Y, "%;\n      transform: translateX(-").concat(PERCENTAGE_X, "%);\n    "),
    right: "\n      left: ".concat(PERCENTAGE_Y, "%;\n      top: ").concat(PERCENTAGE_X, "%;\n      transform: translateY(-").concat(PERCENTAGE_X, "%);\n  "),
    left: "\n      right: ".concat(PERCENTAGE_Y, "%;\n      top: ").concat(PERCENTAGE_X, "%;\n      transform: translateY(-").concat(PERCENTAGE_X, "%);\n    "),
    bottom: "\n      left: ".concat(PERCENTAGE_X, "%;\n      top: ").concat(PERCENTAGE_Y, "%;\n      transform: translateX(-").concat(PERCENTAGE_X, "%);\n    ")
  }
};
var _default = placementConfig;
exports.default = _default;
