// @catho/quantum/Grid/sub-components/Container.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _shared = require("./shared");

var _shared2 = require("../../shared");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Container = _styledComponents.default.div.withConfig({
  displayName: "Container",
  componentId: "ka34zl-0"
})(["box-sizing:border-box;margin-left:auto;margin-right:auto;padding:var(--gutter);", " ", " ", " & &{max-width:100%;}"], function (_ref) {
  var fluid = _ref.fluid;
  return !fluid && 'max-width: 95%;';
}, _shared.hide, _shared.noGutters);

Container.propTypes = {
  fluid: _propTypes.default.bool,
  hide: _propTypes.default.oneOfType([_propTypes.default.oneOf(Object.keys(_shared2.theme.breakpoints)), _propTypes.default.arrayOf(_propTypes.default.string)]),
  theme: _propTypes.default.shape({
    breakpoints: _propTypes.default.object
  }),
  'no-gutters': _propTypes.default.bool
};
Container.defaultProps = {
  fluid: false,
  'no-gutters': false,
  theme: _shared2.theme
};
Container.displayName = 'Container';
var _default = Container;
exports.default = _default;
