// @catho/quantum/Grid/sub-components/Hide.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _media = require("./shared/media");

var _theme = _interopRequireDefault(require("../../shared/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Hide = _styledComponents.default.div.withConfig({
  displayName: "Hide",
  componentId: "sc-13m0had-0"
})(["", ""], function (_ref) {
  var breakpoints = _ref.theme.breakpoints,
      xsmall = _ref.xsmall,
      small = _ref.small,
      medium = _ref.medium,
      large = _ref.large,
      xlarge = _ref.xlarge;
  var style = '';
  var hideBreakpoint = (0, _media.hideQueries)(breakpoints);
  style += xsmall ? hideBreakpoint.xsmall() : '';
  style += xlarge ? hideBreakpoint.xlarge() : '';
  style += small ? hideBreakpoint.small() : '';
  style += medium ? hideBreakpoint.medium() : '';
  style += large ? hideBreakpoint.large() : '';
  return style;
});

Hide.defaultProps = {
  theme: _theme.default
};
Hide.propTypes = {
  xlarge: _propTypes.default.bool,
  large: _propTypes.default.bool,
  medium: _propTypes.default.bool,
  small: _propTypes.default.bool,
  xsmall: _propTypes.default.bool,
  theme: _propTypes.default.shape({
    breakpoints: _propTypes.default.object
  })
};
Hide.displayName = 'Hide';
var _default = Hide;
exports.default = _default;
