// @catho/quantum/Popover/arrowProperties.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _Colors = _interopRequireDefault(require("../Colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ARROW_SIZE = '16px';
var SIDE_POSITION_X = -8;
var CENTER_POSITION = '50%';

var getColorBySkin = function getColorBySkin(skin) {
  var indexColor = skin.toUpperCase();
  return skin === 'default' ? _Colors.default.WHITE : _Colors.default[indexColor][200];
};

var placementPosition = {
  top: "\n    content: '\u25BC';\n    left: ".concat(CENTER_POSITION, ";\n    text-shadow: 0px 4px 4px ").concat(_Colors.default.SHADOW[50], ";\n    bottom: 0;\n    transform: translateX(-").concat(CENTER_POSITION, ");\n  "),
  bottom: "\n    top: -4px;\n    content: '\u25B2';\n    left: ".concat(CENTER_POSITION, ";\n    text-shadow: 0px -3px 4px ").concat(_Colors.default.SHADOW[50], ";\n    transform: translateX(-").concat(CENTER_POSITION, ");\n  "),
  right: "\n    bottom: ".concat(CENTER_POSITION, ";\n    content: '\u25C0';\n    left: ").concat(SIDE_POSITION_X, "px;\n    text-shadow: -3px 1px 4px ").concat(_Colors.default.SHADOW[50], ";\n  "),
  left: "\n    bottom: ".concat(CENTER_POSITION, ";\n    content: '\u25B6';\n    right: ").concat(SIDE_POSITION_X, "px;\n    text-shadow: 3px 1px 4px ").concat(_Colors.default.SHADOW[50], ";\n  ")
};

var getArrow = function getArrow(placement, skin) {
  return "\n  color: ".concat(getColorBySkin(skin), "\n  font-size: ").concat(ARROW_SIZE, ";\n  position: absolute;\n\n  ").concat(placementPosition[placement], "\n");
};

var _default = getArrow;
exports.default = _default;
