// @catho/quantum/Popover/Popover.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireWildcard(require("prop-types"));

var _Content = _interopRequireDefault(require("./sub-components/Content"));

var _options = _interopRequireDefault(require("./options"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "Popover__Wrapper",
  componentId: "sc-94rrmp-0"
})(["display:inline-block;position:relative;"]);

var TriggerBlock = _styledComponents.default.div.withConfig({
  displayName: "Popover__TriggerBlock",
  componentId: "sc-94rrmp-1"
})(["cursor:pointer;"]);

var Popover =
/*#__PURE__*/
function (_Component) {
  _inherits(Popover, _Component);

  function Popover(props) {
    var _this;

    _classCallCheck(this, Popover);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Popover).call(this, props));

    _this.handleVisible = function (visible) {
      var onClose = _this.props.onClose;

      if (!visible) {
        onClose();
      }

      _this.setState({
        visible: visible
      }, function () {
        var newVisibleValue = _this.state.visible;

        if (newVisibleValue) {
          _this.setPopoverPosition();
        }
      });
    };

    _this.setPopoverPosition = function () {
      var placement = _this.props.placement;
      var _this$wrapperRef$curr = _this.wrapperRef.current,
          popoverWrapperTopValue = _this$wrapperRef$curr.offsetTop,
          triggerLeftValue = _this$wrapperRef$curr.offsetLeft,
          triggerWidthValue = _this$wrapperRef$curr.offsetWidth,
          triggerHeightValue = _this$wrapperRef$curr.offsetHeight;
      var _this$contentRef = _this.contentRef,
          innerContentRef = _this$contentRef.innerContentRef,
          _this$contentRef$inne = _this$contentRef.innerContentRef,
          popoverContentWidth = _this$contentRef$inne.offsetWidth,
          popoverContentHeight = _this$contentRef$inne.offsetHeight;
      var position = (0, _options.default)({
        popoverWrapperTopValue: popoverWrapperTopValue,
        popoverContentWidth: popoverContentWidth,
        popoverContentHeight: popoverContentHeight,
        triggerLeftValue: triggerLeftValue,
        triggerWidthValue: triggerWidthValue,
        triggerHeightValue: triggerHeightValue
      });
      innerContentRef.style.left = "".concat(position[placement].left, "px");
      innerContentRef.style.top = "".concat(position[placement].top, "px");

      if (placement === 'right' || placement === 'left') {
        innerContentRef.style.transform = 'translateY(-50%)';
      }
    };

    _this.wrapperRef = _react.default.createRef();
    _this.state = {
      visible: undefined
    };
    return _this;
  }

  _createClass(Popover, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.isVisible) {
        this.setPopoverPosition();
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      if (this.isVisible) {
        this.setPopoverPosition();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          children = _this$props.children,
          trigger = _this$props.trigger,
          placement = _this$props.placement,
          visible = _this$props.visible,
          rest = _objectWithoutProperties(_this$props, ["children", "trigger", "placement", "visible"]);

      return _react.default.createElement(Wrapper, {
        ref: this.wrapperRef
      }, this.isVisible && _react.default.createElement(_Content.default, _extends({
        placement: placement,
        visible: visible,
        onPopoverClose: function onPopoverClose() {
          return _this2.handleVisible(false);
        },
        ref: function ref(element) {
          _this2.contentRef = element;
        }
      }, rest), children), _react.default.createElement(TriggerBlock, {
        onClick: function onClick() {
          return _this2.handleVisible(true);
        }
      }, trigger));
    }
  }, {
    key: "isVisible",
    get: function get() {
      var visibleState = this.state.visible;
      var visible = this.props.visible;
      return visibleState !== undefined ? visibleState : visible;
    }
  }]);

  return Popover;
}(_react.Component);

TriggerBlock.displayName = 'TriggerBlock';
Popover.propTypes = {
  visible: _propTypes.default.bool,
  children: _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.node), _propTypes.default.node]).isRequired,
  placement: _propTypes.default.oneOf(['top', 'right', 'bottom', 'left']),
  skin: (0, _propTypes.oneOf)(['default', 'success', 'warning', 'error']),
  trigger: _propTypes.default.node.isRequired,
  onClose: _propTypes.default.func
};
Popover.defaultProps = {
  visible: false,
  skin: 'default',
  placement: 'top',
  onClose: function onClose() {}
};
var _default = Popover;
exports.default = _default;
