// @catho/quantum/RadioGroup/RadioGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _shared = require("../shared");

var _theme = require("../shared/theme");

var _Radio = _interopRequireDefault(require("./Radio"));

var _RadioButton = _interopRequireDefault(require("./RadioButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Group = (0, _styledComponents.default)(_shared.FieldGroup).attrs({
  role: 'radiogroup'
}).withConfig({
  displayName: "RadioGroup__Group",
  componentId: "sc-1l7wvp0-0"
})(["position:relative;"]);
var ErrorLabel = (0, _styledComponents.default)(_shared.ErrorMessage).withConfig({
  displayName: "RadioGroup__ErrorLabel",
  componentId: "sc-1l7wvp0-1"
})(["margin-left:", "px;"], function (_ref) {
  var xxsmall = _ref.theme.spacing.xxsmall;
  return -xxsmall;
});
ErrorLabel.displayName = 'ErrorLabel';

var RadioGroup = function RadioGroup(_ref2) {
  var type = _ref2.type,
      children = _ref2.children,
      error = _ref2.error,
      inline = _ref2.inline,
      name = _ref2.name,
      onChange = _ref2.onChange,
      options = _ref2.options,
      value = _ref2.value,
      theme = _ref2.theme,
      rest = _objectWithoutProperties(_ref2, ["type", "children", "error", "inline", "name", "onChange", "options", "value", "theme"]);

  var commonProps = {
    name: name,
    error: Boolean(error),
    onChange: onChange,
    inline: inline
  };
  var radioOptions = options.map(function (option) {
    return Object.assign({}, option, _objectSpread({
      key: option.value,
      checked: option.value === value ? true : undefined
    }, commonProps));
  });
  var items = _react.default.Children.map(children, function (child) {
    return _react.default.cloneElement(child, _objectSpread({
      checked: child.props.value === value ? true : undefined
    }, commonProps));
  }) || radioOptions.map(function (props) {
    return type === 'button' ? _react.default.createElement(_RadioButton.default, props) : _react.default.createElement(_Radio.default, props);
  });
  return _react.default.createElement(Group, rest, items, error && _react.default.createElement(ErrorLabel, {
    theme: theme
  }, error));
};

RadioGroup.Radio = _Radio.default;
RadioGroup.Button = _RadioButton.default;
RadioGroup.propTypes = {
  type: _propTypes.default.oneOf(['radio', 'button']),
  options: _propTypes.default.arrayOf(_propTypes.default.shape({
    label: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.node]),
    value: _propTypes.default.string.isRequired,
    disabled: _propTypes.default.bool
  })),
  children: _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.element), _propTypes.default.element]),
  inline: _propTypes.default.bool,
  onChange: _propTypes.default.func,

  /** Initialize RadioGroup with a value */
  value: _propTypes.default.string,
  name: _propTypes.default.string.isRequired,
  error: _propTypes.default.string,
  theme: _propTypes.default.shape({
    colors: _propTypes.default.object,
    spacing: _propTypes.default.object
  })
};
/**
 * Group for Radio components.
 */

RadioGroup.defaultProps = {
  type: 'radio',
  children: undefined,
  error: undefined,
  inline: false,
  onChange: function onChange() {},
  options: [],
  value: undefined,
  theme: {
    colors: _theme.colors,
    spacing: _theme.spacing
  }
};
var _default = RadioGroup;
exports.default = _default;
