// @catho/quantum-storybook-ui/dist/ui/Colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Colors =
/*#__PURE__*/
function () {
  function Colors() {
    _classCallCheck(this, Colors);
  }

  _createClass(Colors, null, [{
    key: "hexToRgba",
    value: function hexToRgba(color, opacity) {
      var R = parseInt(color.substring(1, 3), 16);
      var G = parseInt(color.substring(3, 5), 16);
      var B = parseInt(color.substring(5, 7), 16);
      return "rgba(".concat(R, ", ").concat(G, ", ").concat(B, ", ").concat(opacity, ")");
    }
  }]);

  return Colors;
}();

_defineProperty(Colors, "grey", {
  light: '#eceff1',
  medium: '#d1d8dc',
  harder: '#7b909c',
  strong: '#526977',
  athens: '#f3f3f5'
});

_defineProperty(Colors, "blue", {
  light: '#03a9f4',
  sanjuan: '#294661',
  bayoux: '#546B81',
  curious: '#1a82e2',
  selago: '#e8f2fc'
});

_defineProperty(Colors, "black", '#000000');

_defineProperty(Colors, "white", '#FFFFFF');

_defineProperty(Colors, "pink", {
  amaranth: '#e01e5a'
});

var _default = Colors;
exports.default = _default;
