// @catho/quantum/Pagination/sub-components/ActionButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../../Colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var StyledButton = _styledComponents.default.a.withConfig({
  displayName: "ActionButton__StyledButton",
  componentId: "sc-2etngu-0"
})(["border:1px solid transparent;border-radius:10px;cursor:pointer;font-weight:normal;margin:0 4px;outline:none;padding:8px;text-align:center;text-decoration:none;transition:color 0.2s ease-in-out,background-color 0.2s ease-in-out,border 0.2s ease-in-out;user-select:none;white-space:nowrap;", ""], function (_ref) {
  var ariaDisabled = _ref['aria-disabled'];
  return ariaDisabled ? "\n        color: ".concat(_Colors.default.BLACK['200'], ";\n        cursor: not-allowed;\n      ") : "\n        :hover,\n        :focus {\n          background-color: ".concat(_Colors.default.BLUE['200'], ";\n          border: 1px solid ").concat(_Colors.default.BLUE['500'], ";\n          color: ").concat(_Colors.default.BLUE['500'], ";\n        }\n  ");
});

var ActionButton = function ActionButton(_ref2) {
  var onClick = _ref2.onClick,
      props = _objectWithoutProperties(_ref2, ["onClick"]);

  return _react.default.createElement(StyledButton, _extends({
    onClick: onClick,
    skin: "link"
  }, props));
};

ActionButton.propTypes = {
  ariaDisabled: _propTypes.default.bool,
  onClick: _propTypes.default.func
};
ActionButton.defaultProps = {
  ariaDisabled: false,
  onClick: function onClick() {}
};
var _default = ActionButton;
exports.default = _default;
