// @catho/quantum/Pagination/sub-components/Page.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../../Colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var PageButton = _styledComponents.default.a.withConfig({
  displayName: "Page__PageButton",
  componentId: "pz1cuj-0"
})(["align-items:center;background-color:", ";border:1px solid ", ";border-radius:10px;box-sizing:border-box;color:", ";cursor:pointer;display:flex;font-size:16px;font-weight:", ";justify-content:center;margin:0 4px;min-height:40px;min-width:40px;outline:none;padding:5px 8px;text-decoration:none;transition:background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,color 0.2s ease-in-out,font-weight 0.2s ease-in-out;user-select:none;", ";:hover,:focus{background-color:", ";border-color:", ";color:", ";}"], function (_ref) {
  var active = _ref.active;
  return active ? _Colors.default.BLUE['500'] : _Colors.default.WHITE;
}, function (_ref2) {
  var active = _ref2.active;
  return active ? _Colors.default.BLUE['500'] : _Colors.default.BLACK['200'];
}, _Colors.default.BLACK['700'], function (_ref3) {
  var active = _ref3.active;
  return active ? '700' : 'normal';
}, function (_ref4) {
  var active = _ref4.active;
  return active && "color: ".concat(_Colors.default.WHITE, ";");
}, function (_ref5) {
  var active = _ref5.active;
  return active ? _Colors.default.COBALT['500'] : _Colors.default.BLUE['200'];
}, function (_ref6) {
  var active = _ref6.active;
  return active ? _Colors.default.COBALT['500'] : _Colors.default.BLUE['500'];
}, function (_ref7) {
  var active = _ref7.active;
  return active ? _Colors.default.WHITE : _Colors.default.BLUE['500'];
});

var Page = function Page(_ref8) {
  var children = _ref8.children,
      onClick = _ref8.onClick,
      props = _objectWithoutProperties(_ref8, ["children", "onClick"]);

  return _react.default.createElement(PageButton, _extends({
    onClick: onClick
  }, props), children);
};

Page.displayName = 'Page';
Page.propTypes = {
  active: _propTypes.default.bool,
  children: _propTypes.default.number.isRequired,
  onClick: _propTypes.default.func
};
Page.defaultProps = {
  active: false,
  onClick: function onClick() {}
};
var _default = Page;
exports.default = _default;
