// @catho/quantum/Input/InputTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Input = _interopRequireDefault(require("./Input"));

var _masks = _interopRequireDefault(require("../shared/masks"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var InputTypes = {
  CEP: function CEP(props) {
    return _react.default.createElement(_Input.default, _extends({
      mask: _masks.default.cep
    }, props));
  },
  CNPJ: function CNPJ(props) {
    return _react.default.createElement(_Input.default, _extends({
      mask: _masks.default.cnpj
    }, props));
  },
  CPF: function CPF(props) {
    return _react.default.createElement(_Input.default, _extends({
      mask: _masks.default.cpf
    }, props));
  },
  Date: function Date(props) {
    return _react.default.createElement(_Input.default, _extends({
      mask: _masks.default.date
    }, props));
  },
  Phone: function Phone(props) {
    return _react.default.createElement(_Input.default, _extends({
      mask: _masks.default.phone
    }, props));
  },
  Password: function Password(props) {
    return _react.default.createElement(_Input.default, _extends({
      type: "password"
    }, props));
  }
};
InputTypes.CEP.displayName = 'Input.CEP';
InputTypes.CNPJ.displayName = 'Input.CNPJ';
InputTypes.CPF.displayName = 'Input.CPF';
InputTypes.Date.displayName = 'Input.Date';
InputTypes.Phone.displayName = 'Input.Phone';
InputTypes.Password.displayName = 'Input.Password';
var _default = InputTypes;
exports.default = _default;
