// @catho/quantum/List/sub-components/Content.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Header = _interopRequireDefault(require("./Header"));

var _SubHeader = _interopRequireDefault(require("./SubHeader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var ContentBox = _styledComponents.default.div.withConfig({
  displayName: "Content__ContentBox",
  componentId: "sc-1fmj11p-0"
})(["display:flex;flex-direction:column;"]);

ContentBox.displayName = 'Content';

var Content =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Content, _React$Component);

  function Content() {
    var _getPrototypeOf2;

    var _temp, _this;

    _classCallCheck(this, Content);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _possibleConstructorReturn(_this, (_temp = _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Content)).call.apply(_getPrototypeOf2, [this].concat(args))), _this._renderContent = function (content) {
      var _this$props = _this.props,
          propsHeader = _this$props.header,
          propsSubHeader = _this$props.subheader;

      if (typeof content === 'string') {
        return _react.default.createElement(_Header.default, null, content);
      }

      var _content$header = content.header,
          header = _content$header === void 0 ? propsHeader : _content$header,
          _content$subheader = content.subheader,
          subheader = _content$subheader === void 0 ? propsSubHeader : _content$subheader;
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_Header.default, null, header), subheader && _react.default.createElement(_SubHeader.default, null, subheader));
    }, _temp));
  }

  _createClass(Content, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          content = _this$props2.content,
          children = _this$props2.children;
      return _react.default.createElement(ContentBox, null, children || this._renderContent(content));
    }
  }]);

  return Content;
}(_react.default.Component);

Content.defaultProps = {
  subheader: '',
  header: '',
  content: {},
  children: null
};
Content.propTypes = {
  header: _propTypes.default.string,
  subheader: _propTypes.default.string,
  content: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.shape({
    header: _propTypes.default.string,
    subheader: _propTypes.default.string
  })]),
  children: _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.node), _propTypes.default.node])
};
Content.displayName = 'List.Content';
var _default = Content;
exports.default = _default;
