// @catho/quantum-storybook-ui/dist/components/Heading/Heading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _StoryContainer = _interopRequireDefault(require("../StoryContainer"));

var _Title = _interopRequireDefault(require("../Title"));

var _ViewOnRemote = _interopRequireDefault(require("../ViewOnRemote"));

var _Colors = _interopRequireDefault(require("../../ui/Colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  margin-right: 20px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  font-size: 17px;\n  letter-spacing: initial;\n  line-height: initial;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  background-color: ", ";\n  padding: 0;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var HeadingWrapper = _styledComponents.default.div(_templateObject(), _Colors.default.grey.athens);

var StyledP = _styledComponents.default.p(_templateObject2());

var Image = _styledComponents.default.img(_templateObject3());

var Heading = function Heading(_ref) {
  var name = _ref.name,
      title = _ref.title,
      image = _ref.image,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["name", "title", "image", "children"]);

  return _react.default.createElement(HeadingWrapper, rest, _react.default.createElement(_StoryContainer.default, null, _react.default.createElement(_Title.default, null, image && _react.default.createElement(Image, {
    alt: title,
    src: image,
    width: "50",
    height: "50"
  }), title || name), _react.default.createElement(StyledP, null, children), name && _react.default.createElement(_ViewOnRemote.default, {
    name: name
  })));
};

Heading.defaultProps = {
  name: '',
  title: ''
};
Heading.propTypes = {
  name: _propTypes.default.string,
  image: _propTypes.default.string,
  title: _propTypes.default.string,
  children: _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.node), _propTypes.default.node])
};
var _default = Heading;
exports.default = _default;
