// @catho/quantum/Card/sub-components/Description.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Content = _styledComponents.default.div.withConfig({
  displayName: "Description__Content",
  componentId: "cg02p4-0"
})(["", ""], function (_ref) {
  var small = _ref.small,
      _ref$theme = _ref.theme,
      baseFontSize = _ref$theme.baseFontSize,
      xsmall = _ref$theme.spacing.xsmall;
  return "\n    font-size: ".concat(small ? baseFontSize * 0.75 : baseFontSize * 0.875, "px;\n    margin: ").concat(xsmall, "px 0 0 0;\n  ");
});

var Description = function Description(_ref2) {
  var props = _extends({}, _ref2);

  return _react.default.createElement(Content, props);
};

Description.displayName = 'Card.Description';
Description.propTypes = {
  /** default `font-size` is `14px`, with `small` prop defined the `font-size` is changed to `12px`. */
  small: _propTypes.default.bool,
  theme: _propTypes.default.shape({
    baseFontSize: _propTypes.default.number,
    spacing: _propTypes.default.object
  })
};
Description.defaultProps = {
  small: false,
  theme: {
    baseFontSize: _theme.baseFontSize,
    spacing: _theme.spacing
  }
};
var _default = Description;
exports.default = _default;
