// @catho/quantum/Card/sub-components/Media.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "Media__Wrapper",
  componentId: "sc-7nlm1k-0"
})(["padding-bottom:56.25%;position:relative;overflow:hidden;", ""], function (_ref) {
  var _ref$theme = _ref.theme,
      neutral300 = _ref$theme.colors.neutral[300],
      small = _ref$theme.spacing.small;
  return "\n    margin-top: ".concat(small, "px;\n    background-color: ").concat(neutral300, ";\n  ");
});

var Image = _styledComponents.default.img.withConfig({
  displayName: "Media__Image",
  componentId: "sc-7nlm1k-1"
})(["position:absolute;top:50%;transform:translateY(-50%);width:100%;"]);

var Media = function Media(_ref2) {
  var className = _ref2.className,
      style = _ref2.style,
      theme = _ref2.theme,
      props = _objectWithoutProperties(_ref2, ["className", "style", "theme"]);

  return _react.default.createElement(Wrapper, {
    className: className,
    style: style,
    theme: theme
  }, _react.default.createElement(Image, props));
};

Media.displayName = 'Card.Media';
Media.propTypes = {
  className: _propTypes.default.string,
  style: _propTypes.default.objectOf(_propTypes.default.string),
  theme: _propTypes.default.shape({
    colors: _propTypes.default.object,
    spacing: _propTypes.default.object
  })
};
Media.defaultProps = {
  className: undefined,
  style: undefined,
  theme: {
    colors: _theme.colors,
    spacing: _theme.spacing
  }
};
var _default = Media;
exports.default = _default;
