// @catho/quantum/Card/sub-components/Thumbnail.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Image = _styledComponents.default.img.withConfig({
  displayName: "Thumbnail__Image",
  componentId: "sc-18453ay-0"
})(["border-radius:", ";display:inline-block;height:72px;width:72px;", ""], function (_ref) {
  var rounded = _ref.rounded;
  return rounded ? '50%' : '4px';
}, function (_ref2) {
  var rounded = _ref2.rounded,
      neutral300 = _ref2.theme.colors.neutral[300];
  return "\n    background-color: ".concat(neutral300, ";\n    border-radius: ").concat(rounded ? '50%' : '4px', ";\n  ");
});

var Thumbnail = function Thumbnail(_ref3) {
  var props = _extends({}, _ref3);

  return _react.default.createElement(Image, props);
};

Thumbnail.displayName = 'Card.Thumbnail';
Thumbnail.propTypes = {
  src: _propTypes.default.string.isRequired,
  alt: _propTypes.default.string.isRequired,

  /** Display a rounded Thumbnail. */
  rounded: _propTypes.default.bool,
  theme: _propTypes.default.shape({
    colors: _propTypes.default.object
  })
};
Thumbnail.defaultProps = {
  rounded: false,
  theme: {
    colors: _theme.colors
  }
};
var _default = Thumbnail;
exports.default = _default;
