// @catho/quantum/Card/sub-components/Title.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Heading = _styledComponents.default.h2.withConfig({
  displayName: "Title__Heading",
  componentId: "sc-14fvmc0-0"
})(["font-size:", "px;font-weight:600;margin:0;"], function (_ref) {
  var small = _ref.small,
      baseFontSize = _ref.theme.baseFontSize;
  return "".concat(small ? baseFontSize * 1.25 : baseFontSize * 1.5);
});

var Title = function Title(_ref2) {
  var props = _extends({}, _ref2);

  return _react.default.createElement(Heading, props);
};

Title.displayName = 'Card.Title';
Title.propTypes = {
  /** default `font-size` is `24px`, with `small` prop defined the `font-size` is changed to `20px`. */
  small: _propTypes.default.bool,
  theme: _propTypes.default.shape({
    baseFontSize: _propTypes.default.number
  })
};
Title.defaultProps = {
  small: false,
  theme: {
    baseFontSize: _theme.baseFontSize
  }
};
var _default = Title;
exports.default = _default;
