// @catho/quantum/GlobalStyle/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _GlobalStyle.default;
  }
});
Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function get() {
    return _Typography.default;
  }
});
Object.defineProperty(exports, "IconFont", {
  enumerable: true,
  get: function get() {
    return _IconFont.default;
  }
});
Object.defineProperty(exports, "Title", {
  enumerable: true,
  get: function get() {
    return _Title.default;
  }
});
Object.defineProperty(exports, "Subtitle", {
  enumerable: true,
  get: function get() {
    return _Subtitle.default;
  }
});
Object.defineProperty(exports, "Blockquote", {
  enumerable: true,
  get: function get() {
    return _Blockquote.default;
  }
});
Object.defineProperty(exports, "CSSVariables", {
  enumerable: true,
  get: function get() {
    return _CSSVariables.default;
  }
});

var _GlobalStyle = _interopRequireDefault(require("./GlobalStyle"));

var _Typography = _interopRequireDefault(require("./sub-components/Typography"));

var _IconFont = _interopRequireDefault(require("./sub-components/IconFont"));

var _Title = _interopRequireDefault(require("./sub-components/Title"));

var _Subtitle = _interopRequireDefault(require("./sub-components/Subtitle"));

var _Blockquote = _interopRequireDefault(require("./sub-components/Blockquote"));

var _CSSVariables = _interopRequireDefault(require("./sub-components/CSSVariables"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
