// @catho/quantum/RadioGroup/Radio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../Colors"));

var _Label = _interopRequireDefault(require("../shared/Label"));

var _HiddenInput = _interopRequireDefault(require("../shared/HiddenInput"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var RadioMark = _styledComponents.default.span.withConfig({
  displayName: "Radio__RadioMark",
  componentId: "sc-1dy6exu-0"
})(["background-color:", ";border:1.5px solid ", ";border-radius:50%;box-sizing:border-box;display:inline-block;height:16px;margin-right:5px;position:relative;top:4px;width:16px;:after{background-color:", ";border-radius:50%;content:'';display:none;height:50%;left:25%;position:absolute;top:25%;width:50%;}"], _Colors.default.WHITE, _Colors.default.BLACK[400], _Colors.default.BLUE[500]);

var RadioLabel = (0, _styledComponents.default)(_Label.default).withConfig({
  displayName: "Radio__RadioLabel",
  componentId: "sc-1dy6exu-1"
})(["cursor:pointer;user-select:none;", "{:checked{~ ", "{border-color:", ";:after{background-color:", ";display:block;}}}:focus{~ ", "{border-color:", ";box-shadow:0 2px 6px 0 ", ";}}}:hover,:focus{", "{border-color:", ";box-shadow:0 2px 6px 0 ", ";}}", " ", " ", ":checked:disabled ~ ", "{background-color:", ";border-color:", ";box-shadow:inset 0 0 0 3.5px ", ";}"], _HiddenInput.default, RadioMark, _Colors.default.BLUE[500], _Colors.default.BLUE[500], RadioMark, _Colors.default.BLUE[500], _Colors.default.BLUE[50], RadioMark, _Colors.default.BLUE[500], _Colors.default.BLUE[50], function (_ref) {
  var error = _ref.error;
  return error && "\n    ".concat(_HiddenInput.default, " {\n      :checked {\n        ~ ").concat(RadioMark, " {\n          border-color: ").concat(_Colors.default.ERROR[500], ";\n          :after {\n            background-color: ").concat(_Colors.default.ERROR[500], ";\n            display: block;\n          }\n        }\n      }\n      :focus {\n        ~ ").concat(RadioMark, " {\n          border-color: ").concat(_Colors.default.ERROR[500], ";\n          box-shadow: 0 2px 6px 0 ").concat(_Colors.default.ERROR[500], ";\n        }\n      }\n    }\n\n    ").concat(RadioMark, " {\n      border-color: ").concat(_Colors.default.ERROR[500], ";\n    }\n\n    :hover,\n    :focus {\n      ").concat(RadioMark, " {\n        border-color: ").concat(_Colors.default.ERROR[500], ";\n        box-shadow: 0 2px 6px 0 ").concat(_Colors.default.ERROR[500], ";\n      }\n    }\n    ");
}, function (_ref2) {
  var disabled = _ref2.disabled;
  return disabled && "\n    color: ".concat(_Colors.default.BLACK[400], ";\n\n    :hover {\n      cursor: not-allowed;\n    }\n\n    ").concat(_HiddenInput.default, " {\n      :disabled {\n        ~ ").concat(RadioMark, " {\n          background-color: ").concat(_Colors.default.BLACK[200], ";\n        }\n      }\n\n      :checked {\n        ~ ").concat(RadioMark, " {\n          border-color: ").concat(_Colors.default.ERROR[500], ";\n          :after {\n            background-color: ").concat(_Colors.default.ERROR[500], ";\n            display: block;\n          }\n        }\n      }\n\n      :checked:disabled {\n        background-color: ").concat(_Colors.default.BLACK[400], ";\n        border-color: ").concat(_Colors.default.BLACK[400], ";\n        ~ ").concat(RadioMark, ":after {\n          background-color: ").concat(_Colors.default.BLACK[400], ";\n        }\n      }\n\n      :focus {\n        ~ ").concat(RadioMark, " {\n          border-color: ").concat(_Colors.default.BLACK[400], ";\n          box-shadow: none;\n        }\n      }\n    }\n\n    :hover, :focus {\n      ").concat(RadioMark, " {\n        border-color: ").concat(_Colors.default.BLACK[400], ";\n        box-shadow: none;\n      }\n    }\n  ");
}, _HiddenInput.default, RadioMark, _Colors.default.BLACK[400], _Colors.default.BLACK[400], _Colors.default.WHITE);

var Radio = function Radio(_ref3) {
  var children = _ref3.children,
      label = _ref3.label,
      error = _ref3.error,
      disabled = _ref3.disabled,
      _onChange = _ref3.onChange,
      value = _ref3.value,
      rest = _objectWithoutProperties(_ref3, ["children", "label", "error", "disabled", "onChange", "value"]);

  return _react.default.createElement(RadioLabel, {
    error: error,
    disabled: disabled
  }, _react.default.createElement(_HiddenInput.default, _extends({
    type: "radio",
    disabled: disabled,
    onChange: function onChange(e) {
      return _onChange({
        value: value,
        label: label
      }, e);
    },
    value: value
  }, rest)), _react.default.createElement(RadioMark, null), children || label);
};

Radio.displayName = 'RadioGroup.Radio';
Radio.defaultProps = {
  disabled: false,
  error: false,
  children: undefined,
  label: undefined,
  onChange: function onChange() {}
};
Radio.propTypes = {
  disabled: _propTypes.default.bool,
  error: _propTypes.default.bool,
  children: _propTypes.default.string,
  label: _propTypes.default.string,
  value: _propTypes.default.string.isRequired,
  onChange: _propTypes.default.func
};
var _default = Radio;
exports.default = _default;
