// @catho/quantum/TextArea/TextArea.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _shared = require("../shared");

var _uniqId = _interopRequireDefault(require("../shared/uniqId"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var ID_GENERATOR = (0, _uniqId.default)('textarea-');
var DEFAULT_INPUT_STYLE = _shared.INPUT_STYLE.default,
    LABEL_STYLE = _shared.INPUT_STYLE.LABEL_STYLE,
    HELPER_TEXT_STYLE = _shared.INPUT_STYLE.HELPER_TEXT_STYLE,
    REQUIRED_MARK_STYLE = _shared.INPUT_STYLE.REQUIRED_MARK_STYLE,
    ERROR_MESSAGE_STYLE = _shared.INPUT_STYLE.ERROR_MESSAGE_STYLE;

var TextAreaTag = _styledComponents.default.textarea.withConfig({
  displayName: "TextArea__TextAreaTag",
  componentId: "sc-1vkwnym-0"
})(["", ";display:block;min-height:108px;margin-top:8px;transition:border 0.2s ease-in-out,box-shadow 0.2s ease-in-out;"], DEFAULT_INPUT_STYLE);

var TextAreaLabel = (0, _styledComponents.default)(_shared.Label).withConfig({
  displayName: "TextArea__TextAreaLabel",
  componentId: "sc-1vkwnym-1"
})(["", ""], LABEL_STYLE);

var HelperText = _styledComponents.default.span.withConfig({
  displayName: "TextArea__HelperText",
  componentId: "sc-1vkwnym-2"
})(["", ""], HELPER_TEXT_STYLE);

var RequiredMark = _styledComponents.default.em.withConfig({
  displayName: "TextArea__RequiredMark",
  componentId: "sc-1vkwnym-3"
})(["", ""], REQUIRED_MARK_STYLE);

var TextAreaErrorMessage = (0, _styledComponents.default)(_shared.ErrorMessage).withConfig({
  displayName: "TextArea__TextAreaErrorMessage",
  componentId: "sc-1vkwnym-4"
})(["", " ", ""], ERROR_MESSAGE_STYLE, function (_ref) {
  var helperText = _ref.helperText;
  return helperText && "\n    padding-top: 2px;\n  ";
});

var TextArea =
/*#__PURE__*/
function (_React$Component) {
  _inherits(TextArea, _React$Component);

  function TextArea(props) {
    var _this;

    _classCallCheck(this, TextArea);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(TextArea).call(this, props));
    var id = props.id;
    _this._id = id || ID_GENERATOR.next().value;
    return _this;
  }

  _createClass(TextArea, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          label = _this$props.label,
          required = _this$props.required,
          helperText = _this$props.helperText,
          error = _this$props.error,
          id = _this$props.id,
          rest = _objectWithoutProperties(_this$props, ["label", "required", "helperText", "error", "id"]);

      return _react.default.createElement(_shared.FieldGroup, null, label && _react.default.createElement(TextAreaLabel, {
        htmlFor: this._id
      }, label, required && _react.default.createElement(RequiredMark, null, "*")), _react.default.createElement(TextAreaTag, _extends({
        error: error,
        id: this._id,
        required: required
      }, rest)), helperText && _react.default.createElement(HelperText, null, helperText), error && _react.default.createElement(TextAreaErrorMessage, {
        helperText: helperText
      }, error));
    }
  }]);

  return TextArea;
}(_react.default.Component);

TextArea.defaultProps = {
  disabled: false,
  error: '',
  helperText: '',
  label: undefined,
  onChange: function onChange() {},
  placeholder: '',
  required: false,
  value: '',
  id: undefined
};
TextArea.propTypes = {
  disabled: _propTypes.default.bool,
  error: _propTypes.default.string,
  helperText: _propTypes.default.string,
  label: _propTypes.default.string,
  onChange: _propTypes.default.func,
  placeholder: _propTypes.default.string,
  required: _propTypes.default.bool,
  value: _propTypes.default.string,
  id: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number])
};
TextAreaTag.displayName = 'TextAreaTag';
TextAreaLabel.displayName = 'TextAreaLabel';
HelperText.displayName = 'HelperText';
TextAreaErrorMessage.displayName = 'TextAreaErrorMessage';
var _default = TextArea;
exports.default = _default;
