// @catho/quantum/Pagination/sub-components/ActionButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ActionButton = _styledComponents.default.a.withConfig({
  displayName: "ActionButton",
  componentId: "sc-2etngu-0"
})(["border:1px solid transparent;border-radius:10px;cursor:pointer;font-weight:normal;outline:none;text-align:center;text-decoration:none;transition:color 0.2s ease-in-out,background-color 0.2s ease-in-out,border 0.2s ease-in-out;user-select:none;white-space:nowrap;", ""], function (_ref) {
  var _ref$theme = _ref.theme,
      _ref$theme$colors = _ref$theme.colors,
      neutral300 = _ref$theme$colors.neutral[300],
      _ref$theme$colors$pri = _ref$theme$colors.primary,
      primary100 = _ref$theme$colors$pri[100],
      primary500 = _ref$theme$colors$pri[500],
      _ref$theme$spacing = _ref$theme.spacing,
      xsmall = _ref$theme$spacing.xsmall,
      xxsmall = _ref$theme$spacing.xxsmall;
  return "\n    margin: 0 ".concat(xxsmall, "px;\n    padding: ").concat(xsmall, "px;\n\n      &[aria-disabled=true] {\n        color: ").concat(neutral300, ";\n        cursor: not-allowed;\n      }\n\n      &[aria-disabled=false] {\n        :hover,\n        :focus {\n          background-color: ").concat(primary100, ";\n          border: 1px solid ").concat(primary500, ";\n          color: ").concat(primary500, ";\n        }\n      }\n    }\n  ");
});

ActionButton.propTypes = {
  onClick: _propTypes.default.func,
  theme: _propTypes.default.shape({
    colors: _propTypes.default.object,
    spacing: _propTypes.default.object
  })
};
ActionButton.defaultProps = {
  onClick: function onClick() {},
  theme: {
    colors: _theme.colors,
    spacing: _theme.spacing
  }
};
var _default = ActionButton;
exports.default = _default;
