// @catho/quantum/Pagination/sub-components/Dots.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "Dots__Wrapper",
  componentId: "au6qz7-0"
})(["display:inline-block;margin:", ";min-width:22px;"], function (_ref) {
  var xxsmall = _ref.theme.spacing.xxsmall;
  return "0 ".concat(xxsmall, "px");
});

var Dot = _styledComponents.default.span.withConfig({
  displayName: "Dots__Dot",
  componentId: "au6qz7-1"
})(["position:relative;", ""], function (_ref2) {
  var _ref2$theme = _ref2.theme,
      _ref2$theme$spacing = _ref2$theme.spacing,
      xsmall = _ref2$theme$spacing.xsmall,
      medium = _ref2$theme$spacing.medium,
      neutral300 = _ref2$theme.colors.neutral[300];
  return "\n    &, &:after, &:before {\n      background-color: ".concat(neutral300, ";\n      border-radius: 50%;\n      display: inline-block;\n      height: 5px;\n      width: 5px;\n    }\n\n    &:after, &:before {\n      content: '';\n      position: absolute;\n      top: 0;\n    }\n\n    &:after {\n      left: ").concat(xsmall, "px;\n    }\n\n    &:before {\n      left: ").concat(medium, "px;\n    }\n  ");
});

var Dots = function Dots(_ref3) {
  var theme = _ref3.theme,
      props = _objectWithoutProperties(_ref3, ["theme"]);

  return _react.default.createElement(Wrapper, _extends({
    theme: theme
  }, props), _react.default.createElement(Dot, {
    theme: theme
  }));
};

Dots.propTypes = {
  theme: _propTypes.default.shape({
    colors: _propTypes.default.object,
    spacing: _propTypes.default.object
  })
};
Dots.defaultProps = {
  theme: {
    colors: _theme.colors,
    spacing: _theme.spacing
  }
};
var _default = Dots;
exports.default = _default;
