// @catho/quantum/Pagination/sub-components/PageButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("../../shared/theme");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PageButton = _styledComponents.default.a.withConfig({
  displayName: "PageButton",
  componentId: "sc-1ou28ey-0"
})(["align-items:center;border-radius:10px;box-sizing:border-box;cursor:pointer;display:flex;justify-content:center;min-height:40px;min-width:40px;outline:none;text-decoration:none;transition:background-color 0.2s ease-in-out,border-color 0.2s ease-in-out,color 0.2s ease-in-out,font-weight 0.2s ease-in-out;user-select:none;", ""], function (_ref) {
  var active = _ref.active,
      _ref$theme = _ref.theme,
      baseFontSize = _ref$theme.baseFontSize,
      _ref$theme$colors = _ref$theme.colors,
      _ref$theme$colors$neu = _ref$theme$colors.neutral,
      neutral100 = _ref$theme$colors$neu[100],
      neutral300 = _ref$theme$colors$neu[300],
      neutral700 = _ref$theme$colors$neu[700],
      _ref$theme$colors$pri = _ref$theme$colors.primary,
      primary100 = _ref$theme$colors$pri[100],
      primary500 = _ref$theme$colors$pri[500],
      primary900 = _ref$theme$colors$pri[900],
      _ref$theme$spacing = _ref$theme.spacing,
      xxsmall = _ref$theme$spacing.xxsmall,
      xsmall = _ref$theme$spacing.xsmall;
  return "\n    background-color: ".concat(active ? primary500 : neutral100, ";\n    border: 1px solid ").concat(active ? primary500 : neutral300, ";\n    color: ").concat(active ? neutral100 : neutral700, ";\n    font-size: ").concat(baseFontSize, "px;\n    font-weight: ").concat(active ? '700' : 'normal', ";\n    margin: 0 ").concat(xxsmall, "px;\n    padding: ").concat(xxsmall, "px ").concat(xsmall, "px;\n\n    :hover,\n    :focus {\n      background-color: ").concat(active ? primary900 : primary100, ";\n      border-color: ").concat(active ? primary900 : primary500, ";\n      color: ").concat(active ? neutral100 : primary500, ";\n    }\n  ");
});

PageButton.displayName = 'PageButton';
PageButton.propTypes = {
  active: _propTypes.default.bool,
  children: _propTypes.default.number.isRequired,
  onClick: _propTypes.default.func,
  theme: _propTypes.default.shape({
    baseFontSize: _propTypes.default.number,
    colors: _propTypes.default.object,
    spacing: _propTypes.default.object
  })
};
PageButton.defaultProps = {
  active: false,
  onClick: function onClick() {},
  theme: {
    baseFontSize: _theme.baseFontSize,
    colors: _theme.colors,
    spacing: _theme.spacing
  }
};
var _default = PageButton;
exports.default = _default;
