// @catho/quantum/Button/buttonStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ACTIVE_STYLE = exports.FOCUS_STYLE = exports.HOVER_STYLE = exports.SKIN_STYLE = exports.DEFAULT_STYLE = void 0;

var _styledComponents = require("styled-components");

var _skins3 = _interopRequireDefault(require("./skins"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var fontSize = function fontSize(_ref) {
  var size = _ref.size;
  var sizes = {
    xsmall: '12px',
    small: '12px',
    medium: '16px',
    large: '20px',
    xlarge: '24px'
  };
  return "font-size: ".concat(sizes[size] || sizes.medium, ";");
};

var padding = function padding(_ref2) {
  var size = _ref2.size;
  var paddings = {
    xsmall: '0 12px',
    small: '0 12px',
    medium: '0 16px',
    large: '0 16px',
    xlarge: '0 16px'
  };
  return "padding: ".concat(paddings[size] || paddings.medium, ";");
};

var height = function height(_ref3) {
  var size = _ref3.size;
  var heights = {
    xsmall: '24px',
    small: '32px',
    medium: '40px',
    large: '48px',
    xlarge: '56px'
  };
  return "height: ".concat(heights[size] || heights.medium, ";");
};

var _interactiveStyle = function _interactiveStyle(props, interactiveState) {
  var _skins = (0, _skins3.default)(props),
      _skins$interactiveSta = _skins[interactiveState],
      shadow = _skins$interactiveSta.shadow,
      background = _skins$interactiveSta.background,
      border = _skins$interactiveSta.border,
      color = _skins$interactiveSta.color;

  var disabled = props.disabled;
  return !disabled ? "\n    box-shadow: ".concat(shadow, ";\n    background-color: ").concat(background, ";\n    border-color: ").concat(border, ";\n    color: ").concat(color, ";\n  ") : '';
};

var hover = function hover(props) {
  return _interactiveStyle(props, 'hovered');
};

var focus = function focus(props) {
  return _interactiveStyle(props, 'focused');
};

var active = function active(props) {
  return _interactiveStyle(props, 'selected');
};

var skin = function skin(props) {
  var _skins2 = (0, _skins3.default)(props),
      unselected = _skins2.unselected,
      disabled = _skins2.disabled,
      decoration = _skins2.decoration,
      borderRadius = _skins2.borderRadius;

  return (0, _styledComponents.css)(["background-color:", ";border:1.5px solid ", ";box-shadow:", ";color:", ";", " ", ""], props.disabled ? disabled.background : unselected.background, props.disabled ? disabled.border : unselected.border, props.disabled ? disabled.shadow : unselected.shadow, props.disabled ? disabled.color : unselected.color, decoration ? "text-decoration: ".concat(decoration, ";") : '', borderRadius ? "border-radius: ".concat(borderRadius, ";") : '');
};

var DEFAULT_STYLE = (0, _styledComponents.css)(["align-items:center;", " display:flex;", " font-weight:bold;", " justify-content:center;", " letter-spacing:0.2px;", " *:nth-child(2){margin-left:5px;}transition:all 0.2s ease-in-out;", ""], function (props) {
  return "cursor: ".concat(props.disabled ? 'not-allowed' : 'pointer', ";");
}, fontSize, height, padding, function (props) {
  return props.center && "\n    margin-left: auto;\n    margin-right: auto;\n  ";
}, function (props) {
  return props.full && "width: 100%;";
});
exports.DEFAULT_STYLE = DEFAULT_STYLE;
var SKIN_STYLE = (0, _styledComponents.css)(["", ""], skin);
exports.SKIN_STYLE = SKIN_STYLE;
var HOVER_STYLE = (0, _styledComponents.css)(["", ""], hover);
exports.HOVER_STYLE = HOVER_STYLE;
var FOCUS_STYLE = (0, _styledComponents.css)(["", ""], focus);
exports.FOCUS_STYLE = FOCUS_STYLE;
var ACTIVE_STYLE = (0, _styledComponents.css)(["", ""], active);
exports.ACTIVE_STYLE = ACTIVE_STYLE;
var BUTTON_STYLE = (0, _styledComponents.css)(["", " ", ":hover{", "}:focus{", "}:active{", "}"], DEFAULT_STYLE, SKIN_STYLE, HOVER_STYLE, FOCUS_STYLE, ACTIVE_STYLE);
exports.default = BUTTON_STYLE;
