// @catho/quantum/Button/skins.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _Colors = _interopRequireDefault(require("../Colors"));

var _DefaultButton = _interopRequireDefault(require("./models/DefaultButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var skins = {
  primary: new _DefaultButton.default({
    unselected: _Colors.default.BLUE['500'],
    selected: _Colors.default.COBALT['500'],
    focused: _Colors.default.BLUE['500'],
    focusedShadow: '0 2px 6px 0 rgba(19, 85, 208, 0.5);',
    disabled: _Colors.default.BLACK['400'],
    hovered: _Colors.default.COBALT['500'],
    selectedBorder: _Colors.default.COBALT['500']
  }),
  secondary: new _DefaultButton.default({
    unselected: _Colors.default.WHITE,
    unselectedTextColor: _Colors.default.BLUE['500'],
    unselectedBorder: _Colors.default.BLUE['500'],
    selected: _Colors.default.BLUE['200'],
    selectedBorder: _Colors.default.BLUE['500'],
    disabled: _Colors.default.SHADOW['40'],
    disabledBorder: _Colors.default.BLACK['400'],
    disabledTextColor: _Colors.default.BLACK['400'],
    focused: _Colors.default.WHITE,
    focusedBorder: _Colors.default.BLUE['500'],
    focusedShadow: '0 2px 4px 0 rgba(19, 85, 208, 0.5);',
    hovered: _Colors.default.BLUE['200'],
    hoveredBorder: _Colors.default.BLUE['500']
  }),
  action: new _DefaultButton.default({
    unselected: _Colors.default.PINK['500'],
    selected: _Colors.default.PINK['900'],
    selectedBorder: _Colors.default.PINK['900'],
    disabled: _Colors.default.PINK['500'],
    focused: _Colors.default.PINK['500'],
    focusedShadow: "0 2px 6px 0 ".concat(_Colors.default.PINK['500'], ";"),
    hovered: _Colors.default.PINK['900']
  }),
  link: new _DefaultButton.default({
    unselected: 'transparent',
    unselectedTextColor: _Colors.default.BLUE['500'],
    unselectedShadow: 'none',
    selectedShadow: 'none',
    disabled: 'transparent',
    focused: 'transparent',
    decoration: 'underline',
    borderRadius: 'unset'
  })
};

var _default = function _default(_ref) {
  var skin = _ref.skin;
  return skins[skin] || skins.primary;
};

exports.default = _default;
