// @catho/quantum/Tag/Tag.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../Colors"));

var _Icon = _interopRequireDefault(require("../Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var skinFontColors = {
  inverted: _Colors.default.WHITE,
  blue: _Colors.default.BLUE[500],
  success: _Colors.default.SUCCESS[900],
  warning: _Colors.default.WARNING[900],
  error: _Colors.default.ERROR[900],
  'blue-text': _Colors.default.BLUE[500]
};
var skins = {
  default: "background-color: ".concat(_Colors.default.BLACK[100], ";"),
  stroked: "\n    border: 1px solid;\n    background: none;\n  ",
  inverted: "\n    color: ".concat(skinFontColors.inverted, "\n    background-color: ").concat(_Colors.default.BLACK[400], ";\n  "),
  blue: "\n    color: ".concat(skinFontColors.blue, "\n    background-color: ").concat(_Colors.default.BLUE[200], ";\n  "),
  success: "\n    color: ".concat(skinFontColors.success, "\n    background-color: ").concat(_Colors.default.SUCCESS[200], ";\n  "),
  warning: "\n    color: ".concat(skinFontColors.warning, "\n    background-color: ").concat(_Colors.default.WARNING[200], ";\n  "),
  error: "\n    color: ".concat(skinFontColors.error, "\n    background-color: ").concat(_Colors.default.ERROR[200], ";\n  "),
  'blue-text': "\n    color: ".concat(skinFontColors['blue-text'], "\n    background-color: ").concat(_Colors.default.BLACK[100], ";\n  ")
};
var sizes = {
  small: "\n    font-size: 12px;\n    padding-top: 3px;\n    padding-bottom: 3px;\n  ",
  large: "\n    font-size: 18px;\n  "
};

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "Tag__Wrapper",
  componentId: "sc-1i8fike-0"
})(["border-radius:8px;box-sizing:border-box;display:inline-block;margin-right:8px;padding:4px 12px;", " ", " ", ""], function (_ref) {
  var bold = _ref.bold;
  return bold && "font-weight: bold;";
}, function (_ref2) {
  var skin = _ref2.skin;
  return skins[skin];
}, function (_ref3) {
  var size = _ref3.size;
  return sizes[size];
});

var Content = _styledComponents.default.div.withConfig({
  displayName: "Tag__Content",
  componentId: "sc-1i8fike-1"
})(["", ""], function (_ref4) {
  var onClose = _ref4.onClose;
  return onClose && "\n    display: flex;\n    align-items: center;\n  ";
});

var iconSizes = {
  small: '14px',
  medium: '18px',
  large: '20px'
};

var applyIconSize = function applyIconSize(_ref5) {
  var size = _ref5.size;
  return iconSizes[size];
};

var CloseIcon = (0, _styledComponents.default)(_Icon.default).withConfig({
  displayName: "Tag__CloseIcon",
  componentId: "sc-1i8fike-2"
})([""]);

var CloseButton = _styledComponents.default.button.withConfig({
  displayName: "Tag__CloseButton",
  componentId: "sc-1i8fike-3"
})(["background:none;border:none;cursor:pointer;height:", ";margin-left:4px;padding:0;", "{font-size:", ";color:", ";}"], applyIconSize, CloseIcon, applyIconSize, function (_ref6) {
  var skin = _ref6.skin;
  return skinFontColors[skin];
});

CloseButton.displayName = 'CloseButton';

var Tag = function Tag(_ref7) {
  var children = _ref7.children,
      text = _ref7.text,
      onClose = _ref7.onClose,
      rest = _objectWithoutProperties(_ref7, ["children", "text", "onClose"]);

  return _react.default.createElement(Wrapper, rest, _react.default.createElement(Content, {
    onClose: onClose
  }, children || text, onClose && _react.default.createElement(CloseButton, _extends({}, rest, {
    onClick: onClose
  }), _react.default.createElement(CloseIcon, {
    name: "close"
  }))));
};

Tag.propTypes = {
  bold: _propTypes.default.bool,
  children: _propTypes.default.string,

  /** A callback that is called when close button is clicked */
  onClose: _propTypes.default.func,
  size: _propTypes.default.oneOf(['small', 'medium', 'large']),
  skin: _propTypes.default.oneOf(['default', 'stroked', 'inverted', 'blue', 'success', 'warning', 'error', 'blue-text']),
  text: _propTypes.default.string
};
Tag.defaultProps = {
  bold: false,
  children: '',
  onClose: undefined,
  size: 'medium',
  skin: 'default',
  text: 'Tag text'
};
var _default = Tag;
exports.default = _default;
