// @catho/quantum-storybook-ui/dist/components/ViewOnRemote/ViewOnRemote.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _code = _interopRequireDefault(require("../../assets/images/code.svg"));

var _quantum = require("@catho/quantum");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  position: absolute;\n  right: 40px;\n  bottom: 20px;\n  text-decoration: none;\n  cursor: pointer;\n\n  ", " {\n    opacity: 0.4;\n\n    &:hover {\n      opacity: 1;\n    }\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  width: 20px;\n  height: 20px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var baseUrl = 'https://github.com/catho/quantum/tree/master/components/';

var CodeIco = _styledComponents.default.img(_templateObject());

var Link = _styledComponents.default.a(_templateObject2(), CodeIco);

var ViewOnRemote = function ViewOnRemote(_ref) {
  var name = _ref.name;
  return _react.default.createElement(Link, {
    href: "".concat(baseUrl).concat(name),
    target: "_blank"
  }, _react.default.createElement(_quantum.Tooltip, {
    text: "View Source",
    placement: "left"
  }, _react.default.createElement(CodeIco, {
    src: _code.default
  })));
};

ViewOnRemote.propTypes = {
  name: _propTypes.default.string.isRequired
};
var _default = ViewOnRemote;
exports.default = _default;
