// @catho/quantum/Card/sub-components/Thumbnail.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Colors = _interopRequireDefault(require("../../Colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Image = _styledComponents.default.img.withConfig({
  displayName: "Thumbnail__Image",
  componentId: "sc-18453ay-0"
})(["background-color:", ";border-radius:", ";display:inline-block;height:72px;width:72px;"], _Colors.default.BLACK['200'], function (_ref) {
  var rounded = _ref.rounded;
  return rounded ? '50%' : '4px';
});

var Thumbnail = function Thumbnail(_ref2) {
  var props = _extends({}, _ref2);

  return _react.default.createElement(Image, props);
};

Thumbnail.displayName = 'Card.Thumbnail';
Thumbnail.propTypes = {
  src: _propTypes.default.string.isRequired,
  alt: _propTypes.default.string.isRequired,

  /** Display a rounded Thumbnail. */
  rounded: _propTypes.default.bool
};
Thumbnail.defaultProps = {
  rounded: false
};
var _default = Thumbnail;
exports.default = _default;
